File: src/main/java/com/ing/loan/request/services/CustomerService.java
package com.ing.loan.request.services;

import com.ing.loan.request.dto.CustomerRequest;
import com.ing.loan.request.dto.CustomerResponse;
import com.ing.loan.request.exception.CustomerNotFoundException;

import java.util.List;

public interface CustomerService {

	CustomerResponse createCustomer(CustomerRequest customerRequest);

	CustomerResponse getCustomerById(Long id) throws CustomerNotFoundException;

	List<CustomerResponse> getAllCustomers();

	CustomerResponse updateCustomer(Long id, CustomerRequest customerRequest) throws CustomerNotFoundException;

	void deleteCustomer(Long id) throws CustomerNotFoundException;

}

File: frontend/src/app/services/customer.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CustomerRequest, CustomerResponse } from '../models/api.models';
import { environment } from 'src/environment';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {
  private apiUrl = environment.customerServiceUrl;

  constructor(private http: HttpClient) { }

  createCustomer(customer: CustomerRequest): Observable<CustomerResponse> {
    return this.http.post<CustomerResponse>(this.apiUrl, customer);
  }

  getAllCustomers(): Observable<CustomerResponse[]> {
    return this.http.get<CustomerResponse[]>(this.apiUrl);
  }

  getCustomerById(id: number): Observable<CustomerResponse> {
    return this.http.get<CustomerResponse>(`${this.apiUrl}/customerId/${id}`);
  }

  updateCustomer(id: number, customer: CustomerRequest): Observable<CustomerResponse> {
    return this.http.put<CustomerResponse>(`${this.apiUrl}/${id}`, customer);
  }

  deleteCustomer(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

